src/core/thread/mle_router.cpp

```diff
@@ -1192,7 +1192,8 @@ otError MleRouter::HandleAdvertisement(const Message &aMessage, const Ip6::Messa
     routerId = GetRouterId(sourceAddress.GetRloc16());
 
     if (IsFullThreadDevice() &&
-        static_cast<int8_t>(route.GetRouterIdSequence() - mRouterTable.GetRouterIdSequence()) > 0)
+        ((mRouterTable.GetActiveRouterCount() == 0) ||
+         (static_cast<int8_t>(route.GetRouterIdSequence() - mRouterTable.GetRouterIdSequence()) > 0)))
```

Comment:
Just a question, do you think that to prevent assert on line 1278 it is enough to rely on `processRouteTlv` flag and `ProcessRouteTlv` method? Probably it is safe, but i see how it theoretically might fail:

a) if Advertisement has been received from other node than a parent (`processRouteTlv` would be set to false) - though probably it won't happen since we don't have links with other nodes at that stage
b) if Route64 does not include Leader Id - for whatever reason - fuzz tests might trigger this

If there is any chance of that i would replace `assert` with `VerifyOrExit`, else its fine 👍

Reply:
@LuDuda, thanks for pointing that out!  The related code is only used to update the path cost to the leader.  I added a check around that code to avoid processing if the leader is not in the router table.  Please review again.